internals/overlord/servstate/handlers.go

```diff
@@ -400,6 +400,18 @@ func (s *serviceData) startInternal() error {
 	s.cmd.Stdout = logWriter
 	s.cmd.Stderr = logWriter
 
+	// Add WaitDelay to ensure cmd.Wait() returns in a reasonable timeframe if
+	// the goroutines that cmd.Start() uses to copy Stdin/Stdout/Stderr are
+	// blocked when copying due to a sub-subprocess holding onto them. Read
+	// more details in these issues:
```

Comment:
Since the SIGTERM and SIGKILL commands terminate the entire process group, the issue is only occurring when a child process promotes itself to become a process group leader (```setpgid```) or if it comes a new session leader (```setsid```, which implies the previous as well). This is when it holds on to resources, because the termination logic failed to kill it. Not sure it if may be useful to add enough to capture that this is not always an issue.

Reply:
Added - thanks